crates/transport_iroh/src/connection_context.rs

```diff
@@ -140,4 +174,272 @@ impl ConnectionContext {
             None => ConnectionType::None,
         }
     }
+
+    /// Spawns an asynchronous task to continuously read and handle incoming uni-directional
```

Comment:
These connection and stream related functions have been moved from the lib file here. That's a slightly better encapsulation.